### PR TITLE
Internalize some decls and some testing infrastructure

### PR DESCRIPTION
**Pull request by milseman**

### CODE REVIEW

**Comment by milseman** [comment]
@swift-ci please test

File: Sources/_StringProcessing/Executor.swift [hunk 1]
```diff
@@ -21,8 +20,8 @@ public struct Executor {
   }
 
   // FIXME: Public for prototype
```
Comment (rxwei) [inline Sources/_StringProcessing/Executor.swift:22]:
```suggestion
```

**Review by rxwei — APPROVED** [review]
We should also be able to change `struct ReferenceID` to internal.

**Comment by milseman** [comment]
Good point, I'll update with several more I found

**Comment by milseman** [comment]
@swift-ci please test

**Pull request merged**